RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1457 @@
+                auto const noiseTerm = noiseADC * noiseADC + noisePhotoSq + pedestalWidth * pedestalWidth;
+
+                // store to global memory
+                amplitudesForChannel[sampleWithinWindow] = amplitude;
+                noiseTermsForChannel[sampleWithinWindow] = noiseTerm;
+                electronicNoiseTermsForChannel[sampleWithinWindow] = pedestalWidth;
+
+              }  //end sample loop
+            }    // end channel loop
+            alpaka::syncBlockThreads(acc);
+
+            // compute energy using shrMethod0EnergySamplePair
+            for (auto channel : uniform_group_elements_y(acc, group, nchannels)) {
+              auto const gch = channel.global;
+              auto const lch = channel.local;
+              for (auto i_sample : independent_group_elements_x(acc, nsamplesForCompute)) {
+                auto const sampleWithinWindow = i_sample;
+
+                int32_t const soi =
+                    gch < f01HEDigis.size()
+                        ? soiSamples[gch]
+                        : (gch < nchannelsf015 ? f5HBDigis.npresamples()[gch - f01HEDigis.size()] : soiSamples[gch]);
+
+                auto const id = gch < f01HEDigis.size()
+                                    ? f01HEDigis.ids()[gch]
+                                    : (gch < nchannelsf015 ? f5HBDigis.ids()[gch - f01HEDigis.size()]
+                                                           : f3HBDigis.ids()[gch - nchannelsf015]);
+                auto const did = HcalDetId{id};
+                auto const hashedId =
+                    did.subdetId() == HcalBarrel
+                        ? did2linearIndexHB(id, mahi.maxDepthHB(), mahi.firstHBRing(), mahi.lastHBRing(), mahi.nEtaHB())
+                        : did2linearIndexHE(id,
+                                            mahi.maxDepthHE(),
+                                            mahi.maxPhiHE(),
+                                            mahi.firstHERing(),
+                                            mahi.lastHERing(),
+                                            mahi.nEtaHE()) +
+                              mahi.offsetForHashes();
+
+                auto const recoParam1 = recoParamsWithPS.recoParamView().param1()[hashedId];
+                auto const recoParam2 = recoParamsWithPS.recoParamView().param2()[hashedId];
+
+                auto const nsamplesToAdd = recoParam1 < 10 ? recoParam2 : (recoParam1 >> 14) & 0xF;
+                // NOTE: must take soi, as values for that thread are used...
+                // NOTE: does not run if soi is bad, because it does not match any sampleWithinWindow
+                if (sampleWithinWindow == static_cast<unsigned>(soi)) {
+                  auto const method0_energy = shrMethod0EnergyAccum[lch];
+                  auto const val = shrMethod0EnergySamplePair[lch];
+                  int const max_sample = (val >> 32) & 0xffffffff;
+
+                  float const max_energy = uint_as_float(val & 0xffffffff);
```

Comment:
```suggestion
                  float const max_energy = uint_as_float(static_cast<uint32_t>(val & 0xffffffff));
```

is needed together with the other changes above.